reactioncommerce/reaction
# [3.0.0] Fix account invite flow and account email data

Resolves #5798 
Resolves #5763 
Resolves #5238   
Impact: **minor**  
Type: **bugfix**

## Issue
After Reaction Admin was split off from the API in `release-3.0.0` branch
- inviting new admin users no longer works
- some account emails have broken links because they point to resources that are now on a different server.

## Solution
- Creates a new AccountInvites collection to track invited email addresses.
- No longer immediately creates a user record for the invited address
- If an `Account` document is created for an invited email address, automatically add them to the invited permission groups rather than the default group ("customer"), and then delete the AccountInvites doc.

## Breaking changes
From the perspective of the involved users, the flow for inviting admins and accepting the invitations remains the same. The user and account changes behind the scenes are different, so any custom plugins or services relying on the old way may need updates.

## Testing
Prep:
1. Create first admin user on empty db, using register form on Reaction Admin.
2. Create first shop using form shown on Reaction Admin after logging in.
3. Configure emailing for the shop in shop settings. Enter a shop email address and valid SMTP server info.

Invite New
1. On Accounts screen, click Manage Groups and then use the form to invite a new shop manager by email (real email address you can access).
1. Log out
1. Verify that you receive the invite email and it looks correct. (Some text may be missing if you didn't fill out all shop details yet.)
1. Click button in email
1. Verify you go to Reaction Admin URL. Click Register and register with the same email address that was invited.
1. Verify that you have proper access, correct permissions (`roles`) are on your new user doc, and your account has correct `groups` list. In Reaction Admin UI, you may have to refresh once or twice to see the admin UI. This is a separate issue that will be fixed later.

Invite Existing
1. Log out of Reaction Admin.
2. Register a new user with an email address that isn't in the system yet.
3. Verify you receive a welcome email that looks correct, and the new user should have "customer" group permissions.
4. Log out
5. Log in as the first owner user.
6. On Accounts screen, click Manage Groups and then use the form to invite the email address you just created an account for to become a shop manager.
7. Verify that the account now shows as a shop manager in the UI.
8. Log out, and back in with the just-promoted user. Verify you can do shop manager things.
